libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+                    Instant instantTime = Instant.ofEpochMilli(parseDateTime.getMillis());
+                    OffsetDateTime offsetParserDateTime = OffsetDateTime.ofInstant(instantTime, ZoneId.of("UTC"));
+                    if (offsetParserDateTime.isAfter(startDate) || offsetParserDateTime.isEqual(startDate)) {
+                        if (continuationToken != null) {
+                            if (StringUtils.equals(blobItem.getName(), continuationToken)) {
+                                // we found continuation token
+                                continuationToken = null;
+                            }
+                        } else {
+                            blobs.add(blobItem);
+                            if (blobs.size() == pageSize) {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                // Get the continuation token and loop until it is empty.
+                token = blobPage.getContinuationToken();
+            }
+        } while (!StringUtils.isBlank(token) && blobs.size() < pageSize);
+
+        pagedResult.setItems(blobs
+            .stream()
+            .map(bl -> {
+                BlobClient blobClient = containerClient.getBlobClient(bl.getName());
+                return this.getActivityFromBlobClient(blobClient);
+            })
+            .map(CompletableFuture::join).collect(Collectors.toList()));
+
+        if (pagedResult.getItems().size() == pageSize) {
+           pagedResult.setContinuationToken(blobs.stream().reduce((first, second) -> second).get().getName());
+        }
+
+        return CompletableFuture.completedFuture(pagedResult);
+    }
+
+    public CompletableFuture<PagedResult<TranscriptInfo>> listTranscripts(String channelId,
+                                                                          @Nullable String continuationToken) {
+        final int pageSize = 20;
+
+        if (StringUtils.isBlank(channelId)) {
+            throw new IllegalArgumentException("Missing channelId");
+        }
+
+        String token = null;
+
+        List<TranscriptInfo> conversations = new ArrayList<TranscriptInfo>();
+        do {
+            String prefix = String.format("%1$s/", sanitizeKey(channelId));
+            Iterable<PagedResponse<BlobItem>> resultSegment = containerClient.listBlobsByHierarchy(prefix)
+                .iterableByPage(token);
+            token = null;
+            for (PagedResponse<BlobItem> blobPage: resultSegment) {
+                for (BlobItem blobItem: blobPage.getValue()) {
+                    // Unescape the Id we escaped when we saved it
+                    String conversationId = new String();
+                    try {
+                        conversationId = URLDecoder.decode(Arrays.stream(blobItem.getName().split("/"))
+                            .reduce((first, second) -> second).get(), StandardCharsets.UTF_8.name());
```

Comment:
This is not correct, first about the `reduce` to get the last, and also because [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs#L250) gets the last which accomplish the length > 0